paasta_tools/kubernetes/application/controller_wrappers.py

```diff
@@ -281,7 +282,7 @@ def sync_horizontal_pod_autoscaler(self, kube_client: KubeClient) -> None:
         )
         hpa_exists = self.exists_hpa(kube_client)
         # NO autoscaling
-        if not self.should_have_hpa():
+        if not self.should_have_hpa() or autoscaling_is_paused():
```

Comment:
nit: Now that there are multiple reasons for deleting the HPA, I think its worth adding a log line here just to explain why the HPA is gone.